File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/services/commands/CommandHandler.java
package de.vkb.dojo.es.facilityManagement.services.commands;


import de.vkb.dojo.es.facilityManagement.model.command.Command;
import de.vkb.dojo.es.facilityManagement.model.event.Event;

public interface CommandHandler<C extends Command, E extends Event> {
    CommandHandlerResult<C,E> handle(C command);
}

File: mock-human-resources/src/main/kotlin/de/vkb/dojo/es/humanResources/services/commands/CommandHandler.kt
package de.vkb.dojo.es.humanResources.services.commands

import de.vkb.dojo.es.humanResources.model.command.Command
import de.vkb.dojo.es.humanResources.model.event.Event

interface CommandHandler<C: Command, E: Event> {
    fun handle(command: C): CommandHandlerResult<C, E>
}
